pkg/util/proxy/proxy.go

```diff
@@ -35,7 +35,7 @@ func ConnectCluster(ctx context.Context, cluster *clusterapis.Cluster, proxyPath
 		return nil, err
 	}
 
-	location, proxyTransport, err := Location(cluster, tlsConfig)
+	location, proxyTransport, err := Location(cluster, tlsConfig.Clone())
```

Comment:
Can we have a comment to explain why we need to clone the TLSConfig?

Reply:
Added